app/controllers/posts_controller.rb

```diff
@@ -27,27 +27,25 @@ def update
       render 'edit'
     end
   end
-# # search
-#   def search
-#     if query.present?
-#       Post.index
-#     else
-#       scoped
-#     end
-#   end
-
-#   def search
-#     Post.search params[:query]
-#   end
   
+  # def index
+  #   #@posts = Post.all
+  #   if params[:query].present?
+  #     @posts = Post.search(params[:query]).order('views_count DESC , created_at DESC')
+  #   else
+  #     @posts = Post.published.order('views_count DESC , created_at DESC')
+  #   end
+  #   Statistic.increment_counter(:count, Statistic.home_page_view_count.id)
+  # end
+
+
   def index
-    @posts = Post.all
+    @posts = Post.published.order('views_count DESC, created_at DESC')
     if params[:query].present?
-      @posts = Post.search(params[:query]).order('views_count DESC , created_at DESC')
+      @posts = @posts.search(params[:query])
     else
-      @posts = Post.published.order('views_count DESC , created_at DESC')
+      @posts = @posts
```

Comment:
we do not need to add this else condition now as you have collected all the published posts earlier. So we can go away with the else condition.